deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -400,37 +401,41 @@ def instance_assigned_switches(self) -> bool:
 
     def infrasetup_instance(self) -> None:
         """ Handle infrastructure setup for this instance. """
-        # check if fpga node
+
+        metasim_enabled = self.parent_node.metasimulation_enabled
+
         if self.instance_assigned_simulations():
-            # This is an FPGA-host node.
+            # This is a sim-host node.
 
-            # copy fpga sim infrastructure
+            # copy sim infrastructure
             for slotno in range(len(self.parent_node.sim_slots)):
                 self.copy_sim_slot_infrastructure(slotno)
 
-            self.get_and_install_aws_fpga_sdk()
-            # unload any existing edma/xdma/xocl
-            self.unload_xrt_and_xocl()
-            # copy xdma driver
-            self.fpga_node_xdma()
-            # load xdma
-            self.load_xdma()
+            if not metasim_enabled:
```

Comment:
Maybe infrasetup should be factored into two methods, infrasetup_common + infrasetup_fpga, and exposed in the abstract class, something to make it more clear what the contract of the function is.

At the very least you need to update the docstring to explain how this function needs to behave different when running under a metasim.

Reply:
Updated docstring of the abstract method. I don't want to re-factor this until we have another solid implementation to compare with. Also, I don't want to mess with the order of doing the things in this fn so close to a release.